src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
+  ) {
+    super({
+      secretOrKey: Env.JWT.AUTH_TOKEN,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload) {
+    const { userId } = payload;
+    const client = this.redisService.getClient();
+    const jwtToken = await client.get(`jwt:${userId}`);
+    if (!jwtToken) {
+      throw new UnauthorizedException();
+    }
+    return payload;
```

Comment:
PullProofAI

It's recommended to handle errors that might arise during the validation process and provide meaningful feedback to the caller.

```diff
+ try {
+     const { userId } = payload;
+     const client = this.redisService.getClient();
+     const jwtToken = await client.get(`jwt:${userId}`);
+     
+     if (!jwtToken) {
+         throw new UnauthorizedException('Invalid JWT token');
+     }
+     
+     return payload;
+ } catch (error) {
+     // Handle errors that occur during validation
+     throw new UnauthorizedException('JWT validation failed: ' + error.message);
+ }
```

### Summary
The suggested changes aim to improve error handling within the `JwtStrategy` class, ensuring that exceptions are properly managed during construction and validation processes.


<!-- This is an auto-generated comment by PullProofAI -->